bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/handler/ConnectedDriveBridgeHandler.java

```diff
@@ -0,0 +1,192 @@
+import org.openhab.core.types.Command;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ConnectedDriveBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ConnectedDriveBridgeHandler extends BaseBridgeHandler implements StringResponseCallback {
+    private final Logger logger = LoggerFactory.getLogger(ConnectedDriveBridgeHandler.class);
+    private HttpClientFactory httpClientFactory;
+    private BundleContext bundleContext;
+    private VehicleDiscovery discoveryService;
+    private ServiceRegistration<?> discoveryServiceRegstration;
+    private Optional<ConnectedDriveProxy> proxy = Optional.empty();
+    private Optional<ConnectedDriveConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+    private Optional<String> troubleshootFingerprint = Optional.empty();
+
+    private static final String DISCOVERY_FINGERPRINT = "discovery-fingerprint";
+    private ChannelUID discoveryFingerprintChannel;
+
+    public ConnectedDriveBridgeHandler(Bridge bridge, HttpClientFactory hcf, BundleContext bc) {
+        super(bridge);
+        httpClientFactory = hcf;
+        bundleContext = bc;
+        discoveryService = new VehicleDiscovery(this);
+        discoveryServiceRegstration = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
+                new Hashtable<>());
+        discoveryFingerprintChannel = new ChannelUID(bridge.getUID(), DISCOVERY_FINGERPRINT);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (DISCOVERY_FINGERPRINT.equals(channelUID.getIdWithoutGroup()) && command instanceof OnOffType) {
+            if (command.equals(OnOffType.ON)) {
+                logger.warn(
+                        "###### BMW ConnectedDrive Binding - Discovery Troubleshoot Fingerprint Data - BEGIN ######");
+                logger.warn("### Discovery Result ###");
+                logger.warn("{}", getDiscoveryFingerprint());
+                logger.warn("###### BMW ConnectedDrive Binding - Discovery Troubleshoot Fingerprint Data - END ######");
+                updateState(discoveryFingerprintChannel, OnOffType.OFF);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        configuration = Optional.of(getConfigAs(ConnectedDriveConfiguration.class));
+        if (configuration.isPresent()) {
+            proxy = Optional.of(new ConnectedDriveProxy(httpClientFactory, configuration.get()));
+            // give the system some time to create all predefined Vehicles
+            scheduler.schedule(this::requestVehicles, 5, TimeUnit.SECONDS);
```

Comment:
You should save the future returned here and make sure to cancel it during `dispose()`

Reply:
Done